pkg/apis/sources/v1/apiserver_validation_test.go

```diff
@@ -170,6 +172,38 @@ func TestAPIServerValidation(t *testing.T) {
 			},
 		},
 		want: errors.New("missing field(s): owner.kind"),
+	}, {
+		name: "empty resources",
+		spec: ApiServerSourceSpec{
+			EventMode: "Resource",
+			Resources: []APIVersionKindSelector{},
+			SourceSpec: duckv1.SourceSpec{
+				Sink: duckv1.Destination{
+					Ref: &duckv1.KReference{
+						APIVersion: "v1alpha1",
```

Comment:
tiny nit: For these, I'd prefer to use v1 (here and throughout)

Reply:
@vaikas this is done. Let's merge this and I can create a separate PR for checking if the webhook is actually invoked?

Reply:
Sounds good 👍